genie-web/src/main/java/com/netflix/genie/web/tasks/leader/ClusterCheckerTask.java

```diff
@@ -109,70 +121,99 @@ public ClusterCheckerTask(
     @Override
     public void run() {
         log.info("Checking for cluster node health...");
-        final Set<String> badNodes = new HashSet<>();
         this.jobSearchService.getAllHostsWithActiveJobs()
             .stream()
             .filter(host -> !this.hostName.equals(host))
```

Comment:
I wonder about this cause we basically don't check the health of the "leader node" but it itself could be unhealthy. We can either leave this in or implement logic in the LeadershipCoordinator which gives up leadership based on the Health state that way it's always on a healthy node.